operate/webapp/src/main/java/io/camunda/operate/webapp/security/permission/PermissionsService.java

```diff
@@ -0,0 +1,284 @@
+      final AuthorizationChecker authorizationChecker,
+      final TenantsService tenantsService) {
+    this.securityConfiguration = securityConfiguration;
+    this.authorizationChecker = authorizationChecker;
+    this.tenantsService = tenantsService;
+  }
+
+  /**
+   * getProcessDefinitionPermissions
+   *
+   * @param bpmnProcessId bpmnProcessId
+   * @return permissions the user has for the given bpmnProcessId
+   */
+  public Set<String> getProcessDefinitionPermissions(final String bpmnProcessId) {
+    return getResourcePermissions(bpmnProcessId, AuthorizationResourceType.PROCESS_DEFINITION);
+  }
+
+  /**
+   * getDecisionDefinitionPermissions
+   *
+   * @param decisionId decisionId
+   * @return permissions the user has for the given decisionId
+   */
+  public Set<String> getDecisionDefinitionPermissions(final String decisionId) {
+    return getResourcePermissions(decisionId, AuthorizationResourceType.DECISION_DEFINITION);
+  }
+
+  /**
+   * getResourcePermissions
+   *
+   * @param resourceKey resourceKey
+   * @param resourceType resourceType
+   * @return permissions the user has for the given resource
+   */
+  public Set<String> getResourcePermissions(
+      final String resourceKey, final AuthorizationResourceType resourceType) {
+    final Set<String> permissions = new HashSet<>();
+    if (isAuthorized()) {
+      final Set<PermissionType> permissionTypeSet =
+          authorizationChecker.collectPermissionTypes(
+              resourceKey, resourceType, getAuthentication());
+      permissionTypeSet.forEach(p -> permissions.add(p.name()));
+    }
+
+    return permissions;
+  }
+
+  /**
+   * hasPermissionForProcess
+   *
+   * @return true if the user has the given permission for the process
+   */
+  public boolean hasPermissionForProcess(
+      final String bpmnProcessId, final IdentityPermission identityPermission) {
+    return hasPermissionForResource(
+        bpmnProcessId, AuthorizationResourceType.PROCESS_DEFINITION, identityPermission);
+  }
+
+  /**
+   * hasPermissionForDecision
+   *
+   * @return true if the user has the given permission for the decision
+   */
+  public boolean hasPermissionForDecision(
+      final String decisionId, final IdentityPermission identityPermission) {
+    return hasPermissionForResource(
+        decisionId, AuthorizationResourceType.DECISION_DEFINITION, identityPermission);
+  }
+
+  /**
+   * hasPermissionForResource
+   *
+   * @return true if the user has the given permission for the resource
+   */
+  public boolean hasPermissionForResource(
+      final String resourceId,
+      final AuthorizationResourceType resourceType,
+      final IdentityPermission identityPermission) {
+    if (!permissionsEnabled()) {
+      return true;
+    }
+    if (!isAuthorized()) {
+      return false;
+    }
+    if (identityPermission == null) {
+      throw new IllegalStateException("Identity permission cannot be null");
+    }
```

Comment:
🔧 no need for this nullity check, as we completely control the way the method is invoked